src/addrman.h

```diff
@@ -525,19 +527,26 @@ friend class CAddrManTest;
         if (format >= Format::V2_ASMAP) {
             s >> serialized_asmap_checksum;
         }
+        const bool restore_bucketing{nUBuckets == ADDRMAN_NEW_BUCKET_COUNT &&
+                                     serialized_asmap_checksum == supplied_asmap_checksum};
 
         for (auto bucket_entry : bucket_entries) {
             int bucket{bucket_entry.first};
             const int entry_index{bucket_entry.second};
             CAddrInfo& info = mapInfo[entry_index];
+
+            // The entry shouldn't appear in more than
+            // ADDRMAN_NEW_BUCKETS_PER_ADDRESS. If it has already, just skip
+            // this bucket_entry.
+            if (info.nRefCount >= ADDRMAN_NEW_BUCKETS_PER_ADDRESS) continue;
```

Comment:
So this happens if a new version of Bitcoin Core reduced `ADDRMAN_NEW_BUCKETS_PER_ADDRESS`, and in this case we completely forget the extra addrs.

Why do you think it's more reasonable? Assuming addrman is less-than-half-full, there's a good chance this odd addr can find a new place without evicting anything.

Reply:
Because `ADDRMAN_NEW_BUCKETS_PER_ADDRESS` should be an invariant in addrman. No entry should appear in no more than `ADDRMAN_NEW_BUCKETS_PER_ADDRESS` buckets.

Reply:
@naumenkogs if a new Bitcoin Core reduces `ADDRMAN_NEW_BUCKETS_PER_ADDRESS` from 8 to e.g. 6 and this condition becomes true in the new version, then the address is already in 6 buckets in memory. So we would not completely forget it, or do I miss something?